tests/installation/addon_products_sle.pm

```diff
@@ -126,6 +127,21 @@ sub handle_addon {
     }
 }
 
+sub test_addonurl {
+    my $testvalue = get_var('ADDONURL');
```

Comment:
I guess it's worth to checking if ADDONURL is set when we expect it to be set. If now we will fail to set it, we won't even call this method.

Reply:
We are setting ADDONURL only when the test variables ``WORKAROUND_MODULES`` & ``ALL_MODULES`` are set. So the subroutine should be called only under these conditions as long as I have not missed anything.

Reply:
Right, in a normal case, but when we would call `die` https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/products/sle/main.pm#L299 previously, now we will ignore such scenario, even though that's most likely an error that `ADDONURL` was not set in the end.

Reply:
And as we are currently discussing this topic, I have not seen any usage of ``ALL_MODULES`` variable among openQA test suites definitions.

Reply:
In such a case, it should die on L#181

Reply:
Ah, but it won't get there...right.

Reply:
That can easily be the case, as this approach was used to start testing of SLE 15 which didn't work with modules yet, so we had to add all modules as addons, so I guess we can remove this part. But WORKAROUND_MODULES are still used in staging. So we set ADDONURL when: 
```
sle_version_at_least('15') && !check_var('SCC_REGISTER', 'installation') && (get_var('ALL_MODULES') || get_var('WORKAROUND_MODULES'))
```